### DISCUSSION TITLE
Unrecognized Inverter - Solis

### DISCUSSION THREAD

**Discussion by wills106** [wills106/homeassistant-solax-modbus#521] [Q&A]
If you get the following error in your log:
`"unrecognized inverter type - serial number : {your_serial_number_here}"`

Please list below the first 6 digits here including the Inverter type and kW rating and which plugin you got a response with.
I can then add in your Inverter.
**Please do not raise an issue in addition.**

If you get `unrecognized XYZ inverter type - serial number : unknown` you don't have a working Modbus connection.
**Don't raise an issue, this isn't a fault of the Integration.** Either use one of the existing discussions or start a new one.

**Comment by Nixos22** [comment]
> If you get the following error in your log: `"unrecognized inverter type - serial number : {your_serial_number_here}"`
> 
> Please list below the first 6 digits here including the Inverter type and kW rating and which plugin you got a response with. I can then add in your Inverter. Please do not raise an issue in addition.

I have the same problem with solis RHI-6k-48ES-5G, 6kw inverter connected with rs485-we usb key, the first 6 digits of the serial are 114F62.
i used solax modbus integration

**Reply by wills106** [reply]
Just released [2023.09.3](https://github.com/wills106/homeassistant-solax-modbus/releases/tag/2023.09.3) with your Inverter in plugin_solis

**Comment by Nixos22** [comment]
> Just released [2023.09.3](https://github.com/wills106/homeassistant-solax-modbus/releases/tag/2023.09.3) with your Inverter in plugin_solis

Thanks for the plugin update, now  receive 141 entities all working, I have another question regarding this, can I connect two receivers to a single inverter?
I would like to keep the energy community receiver, + my home assistant. Would I like them to work at the same time?

**Reply by wills106** [reply]
Have a look at:
https://github.com/alienatedsec/solis-ha-modbus-cloud
https://github.com/Jumpy07/Solis---SolisCloud-and-Home-Assistant

If you start getting Modbus read errors there isn't anything I can do about it.
Modbus is only meant to work with 1 Master and Solis Inverters aren't fully Modbus compliant anyway.

**Reply by Nixos22** [reply]
I understand, to be honest I don't know how the data is extracted from the energy community device, I only know that it is connected via USB with an rs-485we key

**Reply by Nixos22** [reply]
wills106, congratulations for the excellent work, I managed to integrate both systems with the guide you recommended https://github.com/Jumpy07/Solis---SolisCloud-and-Home-Assistant , thanks again for your dedication to cause.

**Reply by Nixos22** [reply]
I need to integrate two more Solis S5-EH1P6k-L inverters with SN. 103105 and SN. 143105, is it possible?

**Reply by wills106** [reply]
> I need to integrate two more Solis S5-EH1P6k-L inverters with SN. 103105 and SN. 143105, is it possible?

Yes both should already be supported.

**Reply by Nixos22** [reply]
Thanks, I'll test everything over the weekend, at the moment I have a problem to report regarding the charging/discharging of the batteries, the cursor stops at 20 Amp, and it is not possible to increase it.
![Screenshot (77)](https://github.com/user-attachments/assets/6e7a037e-49c5-4d4f-90cc-5f95ba25967d)

**Comment by dverhelst** [comment]
Hi, 

Been using this add-on for a long time now and recently started receiving this error.
My inverter serial starts with 110C82 , device type RHI-3P(6-10)K-HVES-5G
It's connected through modbus over tcp

**Reply by wills106** [reply]
If you have only started getting this have you seen #528 if you are on HA 2023.9.x?

**Reply by dverhelst** [reply]
Correct, it must have something to do with Modbus. I reverted a few versions and will try again later :)
Note that the last HA + the last beta of this add-on did not work for me.
Consider this question answered @wills106 , thank you for your prompt response!

**Comment by funks81** [comment]
Hi - I'm getting the following error in my Log - however everything seems to be working fine - I do get data sent to the integration:

unrecognized SolisDirect inverter type - serial number : unknown
15:49:28 – (ERROR) SolaX Inverter Modbus (custom integration)
SolisDirect: cannot find serial number, even not for other Inverter
15:49:28 – (ERROR) SolaX Inverter Modbus (custom integration)
SolisDirect: reading serial number from address 0x80ec failed; other address may succeed
15:49:28 – (WARNING) SolaX Inverter Modbus (custom integration)

My inverter is a Solis RHI-6k-48ES-5G 6kW first 6 of my serial number are 603105

Thanks

**Reply by wills106** [reply]
Your Inverter is already supported. If you are on HA 2023.9.x the make sure you are on Integration 2023.09.4.

I don't know which Solis dongles work and which don't. A user posted a working one a few posts up. I have also posted a few links that might help above as well.

**Comment by SoHSolar** [comment]
Dear Will,

I have added Solax modbus power to my home assistant and am confused about the settings I need to add. It is asking for Modbus address as a single number and then asking for modbus ip address. I have left the first at 1 and then added the ip address of my wifi dongle (DLS-W). The integration is not reading any devices or entities. Please could you advise on how to get it communicating? 

Thank you very much!

**Reply by alienatedsec** [reply]
> Please could you advise on how to get it communicating?

@SoHSolar DLS-W is not compatible with Modbus and thus not compatible with this integration. 

Read more here https://github.com/alienatedsec/solis-ha-modbus-cloud

**Reply by honikos** [reply]
> DLS-W is not compatible with Modbus and thus not compatible with this integration.

It is compatible with Modbus. I use DLS-W(Wi-Fi) with homeassistant and the https://github.com/StephanJoubert/home_assistant_solarman integration. No cloud, no extra hardware.

**Reply by alienatedsec** [reply]
> I use DLS-W(Wi-Fi) 

@honikos SolarMan is not the same as Modbus, so it's misleading to say it is.

**Reply by honikos** [reply]
I didn't say that. Solarman uses modbus. E.g.  for Solis hybrid: https://www.scss.tcd.ie/Brian.Coghlan/Elios4you/RS485_MODBUS-Hybrid-BACoghlan-201811228-1854.pdf

**Reply by alienatedsec** [reply]
@honikos Solarman-based devices encapsulate the Modbus protocol from the inverter via RS485 into an Ethernet packet that needs translating. Hence, this integration will not work with the DLS-W data logger. 

However, there are other data loggers e.g., DLS-L, S2-WL-ST that encapsulate the Modbus protocol from the inverter via RS485 into a Modbus TCP that is compatible with this integration, and is more standardised with Modbus across the industry.

**Comment by MattSpeke** [comment]
Hi, Is it possible to add support for Solis Inverter Model S5-EH1P5K-L Serial No last 6:

**Reply by alienatedsec** [reply]
You need the first six digits not the last six digits of your serial number.

**Reply by MattSpeke** [reply]
First 6 digits of serial number are 603104. Removed number from first comment.

**Comment by fbf2201** [comment]
Hi,
"unrecognized Solis_Battery inverter type - serial number : 010262xxxxxxxx"
Is it also possible to add the battery inverter Solis RAI-3K-48ES-5G; S/N 010262........  ?

I'm directly connected through a USR-TCP232-410S and readings with Node-RED are working.

Thank you!

**Reply by wills106** [reply]
> add the battery inverter...

Is this an AC Storage Inverter with no PV then?

**Reply by fbf2201** [reply]
Yes, it is. It's a 1-phase AC battery inverter. 
The registers seem to be basically the same.

**Reply by wills106** [reply]
I have released [2023.09.6](https://github.com/wills106/homeassistant-solax-modbus/releases/tag/2023.09.6) but needs a bit of housekeeping as per the release notes.

**Reply by fbf2201** [reply]
Just updated.
It gets recognized now.

Thanks a lot!

**Comment by KamSandhu44** [comment]
I am getting the following errors: -

unrecognized Solis inverter type - serial number : unknown
9:02:32 AM – (ERROR) SolaX Inverter Modbus (custom integration)
Solis: cannot find serial number, even not for other Inverter
9:02:32 AM – (ERROR) SolaX Inverter Modbus (custom integration)

My inverter is Solis S5-EH1P-L and the 1st 6 serial number digits are 603105

It actually works for a while, maybe an hour, then it stops.

Reloading the plugin, or restarting HA brings it back to life for a short peried of time.

Added error from HA

This error originated from a custom integration.

Logger: custom_components.solax_modbus
Source: custom_components/solax_modbus/const.py:219
Integration: SolaX Inverter Modbus (documentation, issues)
First occurred: 06:46:02 (2 occurrences)
Last logged: 06:47:33

Something went wrong reading from modbus
Traceback (most recent call last):
  File "/config/custom_components/solax_modbus/__init__.py", line 436, in read_modbus_data
    res = self.read_modbus_registers_all()
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/solax_modbus/__init__.py", line 545, in read_modbus_registers_all
    self.data[descr.key] = descr.value_function(0, descr, self.data )
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/solax_modbus/const.py", line 219, in value_function_battery_input_solis
    inout = datadict["battery_charge_direction"]
            ~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'battery_charge_direction'

**Reply by MattSpeke** [reply]
I have the same issue. The data is there but it doesn’t update unless I force a restart of the integration. It also reloaded info when I logged into Solis to control inverter. 

My inverter Model S5-EH1P5K-L First 6 digits of serial number is 603104.

**Reply by MattSpeke** [reply]
![image](https://github.com/wills106/homeassistant-solax-modbus/assets/70768574/28f941f1-2e38-480a-b9db-e9e84b53bb22)

**Reply by honikos** [reply]
For ha, use this integration: https://github.com/StephanJoubert/home_assistant_solarman
solarman protocol
You only need the internal or external logger (often shipped with the inverter)
worked for me.

**Reply by MattSpeke** [reply]
Managed to fix by reducing the polling rate of the modbus data. I want modbus control of the inverter for charging the battery at different rates depending on solar forecast and remaining battery power.

**Reply by KamSandhu44** [reply]
Do you have a script to automate the charging?

**Reply by MattSpeke** [reply]
Not yet. First step was to get the data stable and not hanging. I have Octopus Agile so next step is to find or write scripts. I have the below installed so far. 

https://community.home-assistant.io/t/octopus-agile-display-tariff-in-graphs-tables-best-import-export-periods-all-done-using-node-red-and-jsonata/518762

https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy

**Comment by funks81** [comment]
I have the same problem with Solis S6GR1p1.5K-M connected via TCP/Ethernet.  The first 6 digits of the serial are 180102.  I'm using the Solax Modbus integration.

Thanks

**Reply by honikos** [reply]
Give this a try: https://github.com/StephanJoubert/home_assistant_solarman   worked for me with solis Wifi datalogging stick (do not now which version exactly, Label says "wifi en2-8m", I guess it's a Solis DLS-W Datalogger)

**Reply by funks81** [reply]
That’s a different integration - I have a problem with this integration and came on here to get my problem resolved. Your comment isn’t really helpful.

**Comment by bulletcom** [comment]
Solis have today updated my inverter firmware to fix a BMS alarm issue at 100% SOC. Since this I'm no longer seeing updates in HA. 
Inverter serial number first 6 digits: 603105
Firmware version: B90044-020000

Can you advise what I need to do, please?

**Reply by bulletcom** [reply]
Scratch that - reboot HA and it's now back online and picking up the entity values. Thanks :)

**Comment by funks81** [comment]
So - does appear to be working for me - but very unstable - updates often taking 4-5 mins rather than every 15 secs as set. Sent from my iPhoneOn 16 Nov 2023, at 12:24, bulletcom ***@***.***> wrote:﻿
Scratch that - reboot HA and it's now back online and picking up the entity values. Thanks :)

**Comment by Gregy** [comment]
Hi, I have attempted to use the homeassistant-solax-modbus integration but I got:
```
2023-12-17 18:58:43.222 ERROR (MainThread) [custom_components.solax_modbus.plugin_solis] unrecognized Inverter inverter type - serial number : 103306xxxxxxxx
```

Model: Solis S6-EH3P10K-H-EU
Rated power: 10kW

Directly connected via usb rs485 adapter.

**Reply by wills106** [reply]
I have released [2024.01.1](https://github.com/wills106/homeassistant-solax-modbus/releases/tag/2024.01.1)

**Reply by Gregy** [reply]
Thank you! Works like a charm :pray:

**Reply by dxmone** [reply]
@Gregy just a question, I have the same Inverter. Did you connect a RS485 to the USB port and replaced the data logger that came with the inverter? or how did you connect it? The inverter has a dedicated RS485 port, did you use that one?

**Reply by Gregy** [reply]
I did not connect the datalogger to the usb port at all and left it empty. I have connected homeassistant to the dedicated rs485 interface described in the manual.

**Reply by dxmone** [reply]
okey, i am trying to get this to work. But i get no link on the RS485 port. Do you know if you have to enable it or something? I did not find any information in the manual about baud rates etc.

**Reply by Gregy** [reply]
The baud rate is 9600. I don't think it needs to be enabled. But you have to be careful when creating the rj45 cable. You have to use the correct pins.

**Reply by dxmone** [reply]
I got it to work now :)

**Reply by chronvas** [reply]
How can I change the baud rate?

**Reply by Gregy** [reply]
You cannot. It is unchangeable on the inverter side. You have to set it to 9600 on hass side during the integration setup.

**Comment by frankssciencemonster** [comment]
I have a Solis RHI-1P5K-HVES-5G 5kW inverter, connected with a USR-W610 (modbus to wifi), using the solis profile, and I get: "unrecognized Solis inverter type - serial number : 140C02..."

Thanks for making this plugin and thanks for any help.

**Reply by wills106** [reply]
I have added in your Inverter with [2024.01.6](https://github.com/wills106/homeassistant-solax-modbus/releases/tag/2024.01.6)

**Comment by holly1889** [comment]
Hi having the unrecognized solis pv inverter error message

Logger: custom_components.solax_modbus.plugin_solis_old
Source: custom_components/solax_modbus/plugin_solis_old.py:447
Integration: SolaX Inverter Modbus (documentation, issues)
First occurred: 14:51:50 (1 occurrences)
Last logged: 14:51:50

unrecognized solis pv inverter type - serial number : unknown


I have a Solis-1P3K-4G 3kw, dual mppt S/N 160A80

Thanks for making this plugin

**Comment by Michael-0978** [comment]
Hi, I'm getting the same error:
'unrecognized Solis inverter type - serial number : 134F32xxxxxxxx'
My inverter model is RHI-5K-48ES-5G, rated capacity is 5kW and connected directly via modbus RS485.
Thanks!

**Reply by wills106** [reply]
Just added the Inverter in with [2024.01.8](https://github.com/wills106/homeassistant-solax-modbus/releases/tag/2024.01.8)

**Reply by Michael-0978** [reply]
Working now, thanks so much :)

**Comment by suikerman** [comment]
Hi, Is it possible to add support for Solis Inverter Model S5-GR3P12K Serial No 180507

**Reply by wills106** [reply]
Will add it into the next release.

**Reply by wills106** [reply]
Added it into [2024.03.3](https://github.com/wills106/homeassistant-solax-modbus/releases/tag/2024.03.3)
Still need to work on showing only PV relevant entities on PV only Inverters.

**Reply by suikerman** [reply]
I have a S5-GR3P12K s/n 1805070232
But I get this message:
unrecognized solis inverter type - serial number : unknown
solis: cannot find serial number, even not for other Inverter

What can I do to resolve this.
I installed the latest update today.

Greeting
Arthur

**Comment by surdox** [comment]
Hello, would it be possible to support Solis S5-EH1P6K-L 6kW, connected with waveshare module, Serial: 143105
Thank you

**Reply by wills106** [reply]
I have just released [2024.03.7b4](https://github.com/wills106/homeassistant-solax-modbus/releases/tag/2024.03.7b4) with the Serial added in.

**Comment by neilrfuller** [comment]
Hi, I have a twin inverter setup with 2 x Solis S5-EH1P3.6K-L first 6 of serial No 603102 for both. Would use 2 x waveshare modules & 2 instances of solax modbus integration. Could you advise if these inverters can be supported please? Many thanks.

**Reply by wills106** [reply]
The 6031's are already supported.

You should be able to use just one waveshare if you assign a different address to the Inverters. You will need terminating resistors though.

**Reply by neilrfuller** [reply]
That's great many thanks.

**Reply by neilrfuller** [reply]
Hi, I am a network newbie, just looking to make sure I get this right. I need 120Ω resistors attached between RS485 A & B terminals. 2 x Inverters and waveshare device will be connected in parallel, the resistors being needed on the waveshare device connection and the inverter that is "end of line" ie furthest from waveshare? Is there any risk of damage to the inverters if I get this wrong please?

**Reply by wills106** [reply]
You have the resistor locations correct.
Shouldn't be any risk.
If you leave them out you will probably get communication issues.

**Comment by neilrfuller** [comment]
Ok. I’m using JW solars instructions for making connections here 

https://www.youtube.com/watch?v=wEasws0fIVg <https://www.youtube.com/watch?v=wEasws0fIVg&t=816s&ab_channel=JWSolar> &t=816s&ab_channel=JWSolar

 

Got connectors here

https://www.ebay.co.uk/itm/204361478937

 

On the inverters, pretty clear I need to daisy chain RS485 terminals, also Gnd and +5V ?

 

Do you know if its OK to provide a male connector that I can fit one of my existing Solis dataloggers to, so I can get firmware updates and keep access to Solis cloud? (not that Solis cloud is proving of much use).

**Comment by neilrfuller** [comment]
Apologies, please ignore the last email. Since found Alienatedsec’s stuff which gives chapter & verse on Solis inverters & dataloggers. Not only a total networking noob but also a Github noob. Realised that a solis inverter only giving +5v ain’t gonna power the waveshare device which needs +9v so poe version seems like the solution. I’ll get there eventually….

 

From: ***@***.*** ***@***.***> 
Sent: Wednesday, April 3, 2024 5:03 PM
To: 'wills106/homeassistant-solax-modbus' ***@***.***>
Subject: RE: [wills106/homeassistant-solax-modbus] Unrecognized Inverter - Solis (Discussion #521)

 

Ok. I’m using JW solars instructions for making connections here 

https://www.youtube.com/watch?v=wEasws0fIVg <https://www.youtube.com/watch?v=wEasws0fIVg&t=816s&ab_channel=JWSolar> &t=816s&ab_channel=JWSolar

 

Got connectors here

https://www.ebay.co.uk/itm/204361478937

 

On the inverters, pretty clear I need to daisy chain RS485 terminals, also Gnd and +5V ?

 

Do you know if its OK to provide a male connector that I can fit one of my existing Solis dataloggers to, so I can get firmware updates and keep access to Solis cloud? (not that Solis cloud is proving of much use).

**Comment by chickenonaraft88** [comment]
Hi @wills106 could you add my inverter please?
Model: RHI-3K-48ES-5G
First 6 digits: 160F42

Thank you.

**Comment by pcxray** [comment]
> If you get the following error in your log: `"unrecognized inverter type - serial number : {your_serial_number_here}"`
> 
> Please list below the first 6 digits here including the Inverter type and kW rating and which plugin you got a response with. I can then add in your Inverter. **Please do not raise an issue in addition.**
> 
> If you get `unrecognized XYZ inverter type - serial number : unknown` you don't have a working Modbus connection. **Don't raise an issue, this isn't a fault of the Integration.** Either use one of the existing discussions or start a new one.

Hi please could you add my Solis inverter type RHI-3K-48ES-5G, 3Kw, serial 160F422 - I was using the solis plugin. Many thanks

**Comment by fraserallanuk** [comment]
Hello @wills106,

Can you please add my inverter:  It's Hybrid 3.6kW Solis inverter  

2024-09-04 09:19:44.920 ERROR (MainThread) [custom_components.solax_modbus.plugin_solis] unrecognized solax inverter type - serial number : 160F42


I am connecting to the Solis inverter via MODBUS waveshare devices. (confirmed working as I can read and control the inverter using https://github.com/Pho3niX90/solis_modbus)

**Reply by wills106** [reply]
What's the maximum charge / discharge Amps? Is it 62.5A or 100A or something else?

**Reply by fraserallanuk** [reply]
Hello @wills106,

It's 60amp, yes.

**Reply by wills106** [reply]
I have released [2024.09.1](https://github.com/wills106/homeassistant-solax-modbus/releases/tag/2024.09.1) with this Inverter.

**Reply by fraserallanuk** [reply]
Thank you so much @wills106 ! 
@ae-jones @pcxray - Our inverters are now added and working :)

**Comment by Nixos22** [comment]
Hello @wills106, I wanted to congratulate you on the excellent work done with this integration. I need to integrate two more Solis S5-EH1P6k-L inverters with SN. 103105 and SN. 143105, is it possible?

**Comment by martynleigh** [comment]
Hi, please can you add Sofar hyd3600-es. I have the same problem connected via TCP/Ethernet. The first 6 digits of the serial are SM1ES1. I'm using the Solax Modbus integration. I get the error about unrecognised inverter type despite using sofar_old option. Many thanks.

**Reply by martynleigh** [reply]
I'm really struggling to get this to work. I can see six entities but they all state "unknown". However, the serial number has pulled through, so the inverter is making Comms via the waveshare rs485 to Eth(B) converter. I've followed the guides and keep ending up at the same point. Any ideas? (I'm a bit of a novice, but I've done quite a lot with HA, and have the solarman integration working ok).

**Reply by wills106** [reply]
Which plugin returns your Serial? plugin_sofar or plugin_sofar_old ?

**Comment by omenukas** [comment]
[custom_components.solax_modbus.plugin_solis] Solis: cannot find serial number, even not for other Inverter
[custom_components.solax_modbus.plugin_solis] unrecognized Solis inverter type - serial number : unknown

Solis S5-GR3P12K, 12kw inverter connected with Waveshare rs485-ETH, the first 6 digits of the serial are 180507.
i used solax modbus integration

**Reply by wills106** [reply]
> the first 6 digits of the serial are 1805

You Inverter is already supported. If you are getting unknown then it's not communicating properly.

**Comment by somdeths** [comment]
Solis S6-EH1P8K-L-PLUS, 8kW inverter - first 6 digits is 303126

**Comment by leense** [comment]
Hi @wills106,

I’m trying to connect a SolaX ZDNY-TL 15000 inverter via Modbus (RS485 → Elfin EW11-DE Modbus TCP gateway) and getting:

unrecognized inverter type - serial number : Z1531661930006

The connection itself appears to be working — the integration successfully reads a real serial number (Z1531661930006), so this isn’t a comms issue. Serial prefix: Z15316.

Setup:

	•	Inverter: SolaX ZDNY-TL 15000
	•	Connection: RS485 → Elfin EW11-DE HF7221 (Modbus TCP gateway) → SolaX Inverter Modbus integration (TCP)
	•	Baudrate tested: both 19200 and 9600 (the latter works and gives a serial number as a response)

Could this inverter type be added to the plugin? Happy to provide more info/logs if useful.

Thanks for maintaining this integration!

**Comment by edwincardano** [comment]
I have a Solis s6-eh3p10k-h-eu , first 6 digits are 133306, in the solis plugin there was already 103366. 

I changed the 103366 as follow

        elif seriesnumber.startswith("133306"):
            invertertype = HYBRID | X3 | MPPT4  # Hybrid Gen6  10kW - HV

everything is working fine now.

I saw i can do the same in in plugin_solis_fb00.py, I am right that i have to use the fb00 version?

**Reply by edwincardano** [reply]
i discoverd that the difference between 103366 and 113366 is 1 fase or 3 fase inventer
